docs/api.md

```diff
@@ -0,0 +1,253 @@
+# soundslike - API documentation
+Learn how to use the soundslike API!
+
+First. for the sake of this, im going to use soundslike.com as the url, however localhost:8888 might be the url if you're using docker 
```

Comment:
I don't think it really makes sense to include the base URL in the docs - we can include a section at the top with the discussion about the base url and just have each endpoint be "relative".